src/main/resources/archetype-resources/deployment/k8s/branch/all.jsonnet

```diff
@@ -0,0 +1,180 @@
+local vars = std.extVar('rebuyTemplateVariables');
+
+local presets = {
+  production: {
+    replicas: 2,
+    liveTraffic: true,
+    cpuRequests: '100m',
+  },
+  testing: {
+    replicas: 1,
+    liveTraffic: false,
+    cpuRequests: '100m',
+  },
+  staging: {
+    replicas: 2,
+    liveTraffic: true,
+    cpuRequests: '100m',
+  },
+};
+
+local preset = presets[vars.presetName];
+
+local metadata = {
+  name: vars.applicationName,
+  labels: {
+    name: vars.applicationName,
+    'app.kubernetes.io/name': vars.projectName,
+    'app.kubernetes.io/version': vars.release,
+    'rebuy.com/release': vars.release,
+    'rebuy.com/preset': vars.presetName,
+    'rebuy.com/live-traffic': if preset.liveTraffic then 'enabled' else 'disabled',
+    'rebuy.com/toolstack': 'jvm.spring-boot.v2.6',
+    team: '${team}',
+  },
+};
+
+local selector = {
+  name: vars.applicationName,
+};
+
+{
+  apiVersion: 'v1',
+  kind: 'List',
+  items: [
+    {
+      apiVersion: 'apps/v1',
+      kind: 'Deployment',
+      metadata: metadata,
+
+      spec: {
+        revisionHistoryLimit: 5,
+        selector: { matchLabels: selector },
+        replicas: preset.replicas,
+
+        strategy: {
+          rollingUpdate: {
+            maxUnavailable: 0,
+          },
+        },
+
+        template: {
+          metadata: metadata {
+            annotations: {
+              'prometheus.io/scrape': 'true',
+              'prometheus.io/port': '8080',
+              'prometheus.io/path': '/actuator/prometheus',
+              'sidecar.istio.io/inject': 'true',
+            },
+          },
+          spec: {
+            terminationGracePeriodSeconds: 120,
+            serviceAccountName: vars.projectName,
+            topologySpreadConstraints: [
+              {
+                maxSkew: 1,
+                topologyKey: 'topology.kubernetes.io/zone',
+                whenUnsatisfiable: 'ScheduleAnyway',
+                labelSelector: { matchLabels: selector },
+              },
+            ],
+            containers: [
+              {
+                name: 'silo',
+                image: std.format('074509403805.dkr.ecr.eu-west-1.amazonaws.com/%s:%s', [vars.projectName, vars.revision]),
+                imagePullPolicy: 'Always',
+                env: [
+                  {
+                    name: 'SILO_PROFILE',
+                    value: std.join(',', ['kubernetes', vars.clusterProfile]),
+                  },
+                  {
+                    name: 'SILO_JAVA_ARGS',
+                    value: std.join(' ', [
```

Comment:
We might not need the join here anymore, because only two args are left. But I do not have an actual opinion on this,

Reply:
I'd still leave it, who knows what will be inserted invidually later.